drivers/adc/adc_ads1x1x.c

```diff
@@ -463,13 +532,17 @@ static void adc_context_start_sampling(struct adc_context *ctx)
 
 	ads1x1x_start_conversion(data->dev);
 
-	k_sem_give(&data->acq_sem);
+	/* Give semaphore only if the thread is running */
+	if (data->tid) {
+		k_sem_give(&data->acq_sem);
+	}
 }
 
 static int ads1x1x_adc_start_read(const struct device *dev, const struct adc_sequence *sequence)
 {
 	int rc;
 	struct ads1x1x_data *data = dev->data;
+	const struct ads1x1x_config *config = dev->config;
```

Comment:
This would give an unsed warning if CONFIG_ADC_ADS1X1X_TRIGGER is not set

Reply:
i will fix it

Reply:
done